packages/server-common/test/handlers/user/sync-active-campaign-contact-factory.test.ts

```diff
@@ -183,6 +187,48 @@ describe('Sync ActiveCampaign Contact Factory', () => {
     );
   });
 
+  it('should call unsubscribe from lists if user is alumni', async () => {
+    const userId = getUserDataObject().id;
+    const user = {
+      ...getUserResponse(),
+      alumniSinceDate: '2020-09-23T20:45:22.000Z',
+    } as UserResponse;
+    userController.fetchById.mockResolvedValue(user);
+    mockActiveCampaign.getContactIdByEmail.mockResolvedValue(null);
+
+    const event = getEventBridgeEventMock(userId);
+
+    await syncActiveCampaignContactHandler(event);
+
+    expect(
+      mockActiveCampaign.unsubscribeContactFromAllLists,
+    ).toHaveBeenCalledWith(
+      activeCampaignAccount,
+      activeCampaignToken,
+      activeCampaignId,
+    );
+    expect(mockActiveCampaign.addContactToList).not.toHaveBeenCalled();
+  });
+
+  it('should not call unsubscribe from lists if user is not alumni', async () => {
+    const userId = getUserDataObject().id;
+    const user = {
+      ...getUserResponse(),
+      alumniSinceDate: undefined,
+    } as UserResponse;
+    userController.fetchById.mockResolvedValue(user);
+    mockActiveCampaign.getContactIdByEmail.mockResolvedValue(null);
```

Comment:
what about when they were not an alumni but have become one, surely this would then return something and it would be another valid use case that you should test, no? I feel like this test is just a repetition of the one from line 95

Reply:
We always fetch the user from Contentful which is the source of truth. 

<img width="498" alt="Screenshot 2024-02-14 at 10 32 51" src="https://github.com/yldio/asap-hub/assets/16595804/2f28383f-a396-4b01-baae-cb7706cd7c81">

I believe it doesn't matter in this case what comes back from `getContactIdByEmail` from Active Campaign 🤔 because even if the data is not synced yet there, we always have the data from Contentful. I can change the mock in this test though to seem more realistic

Reply:
Committed here [adjustment in test mock](https://github.com/yldio/asap-hub/pull/4157/commits/bcbc666db1dbd50947cc7f2db4ecf870d2678591) :)